pkg/controller/datadogagentdeployment/utils.go

```diff
@@ -1018,10 +1019,74 @@ func getPodAffinity(affinity *corev1.Affinity, labelValue string) *corev1.Affini
 	}
 }
 
+func updateDaemonSetStatus(ds *appsv1.DaemonSet, dsStatus *datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus, updateTime *metav1.Time) *datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus {
+	if dsStatus == nil {
+		dsStatus = &datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus{}
+	}
+	if ds == nil {
+		dsStatus.State = string(datadoghqv1alpha1.DatadogAgentDeploymentStateFailed)
+		return dsStatus
+	}
+	if updateTime != nil {
+		dsStatus.LastUpdate = updateTime
+	}
+
+	dsStatus.CurrentHash = getHashAnnotation(ds.Annotations)
+	dsStatus.Desired = ds.Status.DesiredNumberScheduled
+	dsStatus.Current = ds.Status.CurrentNumberScheduled
+	dsStatus.Ready = ds.Status.NumberReady
+	dsStatus.Available = ds.Status.NumberAvailable
+	dsStatus.UpToDate = ds.Status.UpdatedNumberScheduled
+
+	var deploymentState datadoghqv1alpha1.DatadogAgentDeploymentState
+	if dsStatus.UpToDate != dsStatus.Desired {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateUpdating
+	} else if dsStatus.Ready == 0 {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateProgressing
+	} else {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateRunning
+	}
+
+	dsStatus.State = fmt.Sprintf("%v (%d/%d/%d)", deploymentState, dsStatus.Desired, dsStatus.Ready, dsStatus.UpToDate)
```

Comment:
could you add the pods "available" count too?

Reply:
Can do, though I believe we should pick ready or available, putting 4 indicators may be a big too much. I think ready and available are quite close so any of the 2 should be enough, no?

Reply:
I thought it made sense compared to the current daemonset display.
```
k get ds
NAME            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
datadog-agent   1638      1633      1615    1616         1622        <none>          306d
```
What I want to high light with this info is pod that start properly (ready at least one) but after in crashloop. In this case `available` will be greater than `ready`

Reply:
we can start with what you did. If people needs more info they can look at the DaemonSet

Reply:
> I thought it made sense compared to the current daemonset display.
> 
> ```
> k get ds
> NAME            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
> datadog-agent   1638      1633      1615    1616         1622        <none>          306d
> ```
> 
> What I want to high light with this info is pod that start properly (ready at least one) but after in crashloop. In this case `available` will be greater than `ready`

That's why I kept ready over available, I guess it gives a better idea if everything is working smoothly or not.